### PR TITLE
shadowsocks-libev: bump to version 3.0.6

### PR DESCRIPTION
**Pull request by yousong**
Maintainer: @aa65535 
Compile tested: malta, mips_24kc, LEDE master
Run tested: malta qemu machine with ss-local

### CODE REVIEW

**Comment by diizzyy** [comment]
libudns should be a separate ticket

**Comment by yousong** [comment]
hi, @diizzyy , udns is now a requirement because it is not bundled with the new version of shadowsocks-libev.

**Comment by yousong** [comment]
v2 <- v1 change commit message title from libudns to udns.

**Comment by diizzyy** [comment]
I know, however udns isn't shadowsocks-libev hence it should be a separate pull request.

**Comment by yousong** [comment]
If it's causing inconvenience, please do tell.  I think a pr for a single purpose instead of strictly a single package subdir is totally fine and reasonable.

**Comment by stintel** [comment]
If (lib)udns is now a dependency of shadowsocks-libev, it actually makes more sense to do it in the same PR.

File: libs/udns/Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	CATEGORY:=Libraries
+	TITLE:=DNS Resolver Library
+	URL:=http://www.corpit.ru/mjt/udns.html
+endef
+
+define Package/libudns/description
+  UDNS is a stub DNS resolver library with ability to perform both syncronous
+  and asyncronous DNS queries.
+endef
+
```
Comment (stintel) [inline libs/udns/Makefile]: I would add:
CONFIGURE_ARGS:= \
	$(DISABLE_IPV6)

**Comment by yousong** [comment]
v3 <- v2

 - udns: accomodate --disable-ipv6
 - udns: packaging dnsget, rblcheck, ex-rdns utilities

File: libs/udns/Makefile (hunk 1 above)
Comment (yousong) [inline libs/udns/Makefile]: Fixed now.  Originally I wrote the configure.ac for my own use and omitted the --disable-ipv6 configure option

**Comment by diizzyy** [comment]
If we go by history ~95% or so of all deps have been submitted as separate requests and I think this makes more sense. You can easily link PRs.

File: libs/udns/Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+define Package/libudns/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libudns.so* $(1)/usr/lib/
+endef
+
+define udns-utility-template
+  define Package/udns-$(1)
+    SECTION:=utils
+    CATEGORY:=Utitlity
```
Comment (stintel) [inline libs/udns/Makefile]: Should be "Utilities"

**Comment by yousong** [comment]
v4 <- v3 udns: fix typo of section name to Utilities

I'd like to wait for a few days then do the push anyway.

**Comment by yousong** [comment]
v5 <- v4 udns: remove the trailing "EOF" here document mark in the patch...

**Comment by shounak-de** [comment]
@yousong @stintel Thank you very much! :+1:

**Pull request merged**